test/conftest.py

```diff
@@ -0,0 +1,62 @@
+from importlib.util import find_spec
+
+import pandas as pd
+import pytest
+from narwhals.typing import IntoDataFrame
+
+
+def pandas_constructor(obj) -> IntoDataFrame:
+    return pd.DataFrame(obj)  # type: ignore[no-any-return]
+
+
+# TODO: sklearn.utils.validation.check_array does not support it
+# def pandas_nullable_constructor(obj) -> IntoDataFrame:
+#     return pd.DataFrame(obj).convert_dtypes(dtype_backend="numpy_nullable")  # type: ignore[no-any-return]
+
+# TODO: sklearn.utils.validation.check_array does not support it
+# def pandas_pyarrow_constructor(obj) -> IntoDataFrame:
+#     return pd.DataFrame(obj).convert_dtypes(dtype_backend="pyarrow")  # type: ignore[no-any-return]
```

Comment:
This is an interesing one.

To reproduce:
```python
import pandas as pd
from sklearn.utils.validation import check_array

df = pd.DataFrame({"alpha": ["a", "a", "b"], "beta": [1, 2, 1]}).convert_dtypes(dtype_backend="pyarrow")

check_array(df, dtype=None)
```

Did you encounter this already @Fbruzzesi? Maybe in `scikit-lego`?

Reply:
I believe scikit-lego was very early days of Narwhals, and we are not testing with different pandas `dtype_backend` 🙈